modules/planning/tasks/deciders/open_space_decider/open_space_roi_decider.cc

```diff
@@ -53,29 +55,53 @@ Status OpenSpaceRoiDecider::Process(Frame *frame) {
 
   vehicle_state_ = frame->vehicle_state();
   obstacles_by_frame_ = frame->GetObstacleList();
-  const auto &routing_request = frame->local_view().routing->routing_request();
 
-  // TODO(Jinyun): support options on creating dead end boundary
-  if (routing_request.has_parking_space() &&
-      routing_request.parking_space().has_id()) {
-    target_parking_spot_id_ = routing_request.parking_space().id().id();
+  std::array<Vec2d, 4> spot_vertices;
+  Path nearby_path;
+
+  const auto &roi_type = config_.open_space_roi_decider_config().roi_type();
+  if (roi_type == OpenSpaceRoiDeciderConfig::PARKING) {
+    const auto &routing_request =
+        frame->local_view().routing->routing_request();
+
+    if (routing_request.has_parking_space() &&
+        routing_request.parking_space().has_id()) {
```

Comment:
Not good practice to check `has_field`. `routing_request.parking_space().id().id().empty()` is better.